include/dkm_parallel.hpp

```diff
@@ -77,7 +77,14 @@ std::vector<std::array<T, N>> random_plusplus_parallel(
 
 	for (uint32_t count = 1; count < k; ++count) {
 		// Calculate the distance to the closest mean for each data point
-		auto distances = details::closest_distance_parallel(means, data);
+		std::vector<T> distances = details::closest_distance_parallel(means, data);
+		auto sum_of_elems = std::accumulate(distances.begin(), distances.end(),0.0);
+		if (sum_of_elems == 0)
```

Comment:
This is a minor fix, but let's be accurate. The `std::accumulate()` returned type is deduced by the type of the initial value we supply (here a `0.0`, hence a double). So, the condition at the line 82 should compare with a double too.

```suggestion
		auto sum_of_elems = std::accumulate(distances.begin(), distances.end(), 0.0);
		if (sum_of_elems == 0.0)
```